nixos/modules/services/wayland/hypridle.nix

```diff
@@ -10,17 +10,31 @@ let
 in
 {
   options.services.hypridle = {
-    enable = lib.mkEnableOption "hypridle, Hyprland's idle daemon";
+    enable = lib.mkEnableOption null // {
+      description = ''
+        Whether to enable Hypridle, Hyprland's idle daemon.
+
+        ::: {.note}
+        For this service to work properly on Hyprland, you
+        need to have `programs.hyprland.withUWSM` enabled.
```

Comment:
We should assert that.

Reply:
No, because @fufexan wants it to be compatible with other compositors.

Reply:
As it stands right now, it cannot be compatible with other compositors because it depends on the hyprland-uwsm-specific units.

If you wanted to use it with another compositor, you should bring your own hypridle unit.

Reply:
It could be compatible if you just add more targets in `WantedBy` and `After`.

Reply:
Or simply add `systemd.user.services.swayidle.wantedBy = ["your-compositors-session.target"];`. This should be an admonition in the enable option's description.